app/controllers/users_controller.rb

```diff
@@ -0,0 +1,9 @@
+class UsersController < ApplicationController
+  def show
+    @recipe_foods = RecipeFood.joins(:food, :recipe).where(food: { user: current_user }, recipe: { user: current_user })
+    @items_count = @recipe_foods.select('food_id').distinct.count
+    @total_amount = @recipe_foods.sum('quantity * price')
+    @items = @recipe_foods.group('food.name, price, measurement_unit')
+      .select('food.name, SUM(quantity) as quantity, (sum(quantity) * price) as price, measurement_unit')
+  end
```

Comment:
- [ ] You should avoid making SQL queries from the controller. Your controllers should only redirect data that they get from models. You should always make SQL queries from the model.  
The controllers should be concerned with HTTP verbs. (thin controller rule) By that, I mean that a controller’s responsibility should be focused on web requests. Other operations should be handled on the model.